pallets/ah-migrator/src/multisig.rs

```diff
@@ -61,29 +61,47 @@ impl<T: Config> Pallet<T> {
 	}
 
 	pub fn do_receive_multisig(multisig: RcMultisigOf<T>) -> Result<(), Error<T>> {
-		log::debug!(target: LOG_TARGET, "Integrating multisig {}, {:?}", multisig.creator.to_ss58check(), multisig.deposit);
+		// Translate the creator account from RC to AH format
+		let translated_creator = Self::translate_account_rc_to_ah(multisig.creator.clone());
+
+		// Translate the details account (derived multisig account) if present
+		let translated_details = multisig
+			.details
+			.as_ref()
+			.map(|details_account| Self::translate_account_rc_to_ah(details_account.clone()));
+
+		log::debug!(target: LOG_TARGET, "Integrating multisig {} -> {}, deposit: {:?}, details: {:?} -> {:?}",
+			multisig.creator.to_ss58check(),
+			translated_creator.to_ss58check(),
+			multisig.deposit,
+			multisig.details.as_ref().map(|d| d.to_ss58check()),
+			translated_details.as_ref().map(|d| d.to_ss58check())
+		);
 
 		let missing = <T as pallet_multisig::Config>::Currency::unreserve(
-			&multisig.creator,
+			&translated_creator,
 			multisig.deposit,
 		);
 
 		if missing != Default::default() {
 			if KNOWN_BAD_MULTISIGS.contains(&multisig.creator) {
```

Comment:
You can probably use the new one as well, i think the translation is no-op here.